src/compressor/zstd/CMakeLists.txt

```diff
@@ -1,14 +1,14 @@
 # zstd
 
 # libzstd - build it statically
-set(ZSTD_C_FLAGS -fPIC -Wno-unused-variable -O3)
+set(ZSTD_C_FLAGS -fPIC -Wno-unused-variable -O3 -Wno-implicit-fallthrough)
```

Comment:
@adamemerson `-Wimplicit-fallthrough` is introduced by GCC, will older GCCs or other compilers complain at seeing it? if yes, we need to check it with the used compiler before passing it to it.

Reply:
I'll get rid of this. It doesn't actually work due to zstd (at least the cmake version) adding -Wall and -Wextra AFTER any user supplied flags. The Makefile build allows us to pass flags /after/ whatever ones it wants to use and turn warnings off, but I don't think we want to make that change quite yet, since it doesn't support out-of-tree builds well.